pkg/api/vm/handler.go

```diff
@@ -397,6 +426,97 @@ func (h *vmActionHandler) abortMigration(namespace, name string) error {
 	return nil
 }
 
+func (h *vmActionHandler) findMigratableNodes(rw http.ResponseWriter, namespace, name string) error {
+	vmi, err := h.vmiCache.Get(namespace, name)
+	if err != nil {
+		return err
+	}
+
+	if !canMigrate(vmi) {
+		return errors.New("The VM is not migratable")
+	}
+
+	nodes, err := h.findMigratableNodesByVMI(vmi)
+	if err != nil {
+		return err
+	}
+	resp := FindMigratableNodesOutput{
+		Nodes: nodes,
+	}
+
+	util.ResponseOKWithBody(rw, resp)
+	return nil
+}
+
+func (h *vmActionHandler) findMigratableNodesByVMI(vmi *kubevirtv1.VirtualMachineInstance) ([]string, error) {
+	nodeSelector, err := h.getNodeSelectorRequirementFromVMI(vmi)
+	if err != nil {
+		return nil, err
+	}
+
+	nodes, err := h.nodeCache.List(nodeSelector)
+	if err != nil || len(nodes) == 0 {
+		return nil, err
+	}
+
+	// ignore the node where the VM is running
+	migratableNodes := make([]string, 0, len(nodes)-1)
+	for _, node := range nodes {
+		if vmi.Status.NodeName == node.Name {
+			continue
+		}
+
+		if isDrained(node) {
+			continue
+		}
+
+		migratableNodes = append(migratableNodes, node.Name)
+	}
+	return migratableNodes, nil
+}
+
+func isDrained(node *corev1.Node) bool {
+	if _, ok := node.Annotations[nodecontroller.MaintainStatusAnnotationKey]; ok {
+		return ok
+	}
+	if _, ok := node.Annotations[drainhelper.DrainAnnotation]; ok {
+		return ok
+	}
+	if node.Spec.Unschedulable {
+		return true
+	}
+	if node.Spec.Taints != nil {
+		for _, taint := range node.Spec.Taints {
+			if taint.Key == corev1.TaintNodeUnreachable || taint.Key == corev1.TaintNodeUnschedulable {
+				return true
+			}
+		}
+	}
+
+	return false
+}
+
+func (h *vmActionHandler) getNodeSelectorRequirementFromVMI(vmi *kubevirtv1.VirtualMachineInstance) (labels.Selector, error) {
+	if vmi == nil {
+		return nil, nil
+	}
```

Comment:
```suggestion
	if vmi == nil || vmi.Spec.Affinity == nil || vmi.Spec.Affinity.NodeAffinity == nil || vmi.Spec.Affinity.NodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution == nil {
		return labels.Everything(), nil
	}
```

Reply:
if the vm is using management Network, `vmi.Spec.Affinity.NodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution.NodeSelectorTerms` will panic
and we also need return `labels.Everything()` is this case. if we return `nil`, `nodeCache.List(nodeSelector)` will panic too.